QA.md

```diff
@@ -72,11 +78,40 @@ All of the above tests need to be run on these providers:
 * lxd (time permitting)
 * openstack on bastion[1] or canonistack
 * Azure
-  - Do not test with juju 1.24
-  - juju 1.25 should allow model creation, pending GUI support.
-  - juju 2.0 will not allow model creation in the near term.
+  - Do not test with Juju 1.24
+  - Juju 1.25 should allow model creation, pending GUI support.
+  - Juju 2.0 will not allow model creation in the near term.
+
+## Automated functional testing ##
+
+The uitest project contains integration tests for many Juju services, including
```

Comment:
I know, do you think that's a problem? We need to mention how to run functional tests, and the fact that we have something internally that we use to run integration tests is not really a secret.